compiler/passes/scopeResolve.cpp

```diff
@@ -1410,6 +1411,11 @@ static void resolveModuleCall(CallExpr* call) {
           sym = scope->lookupPublicImports(mbrName);
         }
 
+        // Then, try public import statements that enables unqualified access
+        if (!sym) {
+          sym = scope->lookupPublicUnqualAccessSyms(mbrName, modArg);
```

Comment:
There's a question about ordering of symbols here, which falls into two points.

1. For regular resolution, we look at the symbols for unqualified access first, then look at the modules that are explicitly mentioned in use and import statements, but this code placement has us doing the opposite.
2. This doesn't account for multiple symbols with the same name being brought in, which is something we're not checking for in this re-exporting stuff at all just yet (should have thought about it earlier, sorry).  Right now, you'll only return the first match you find, but there could be multiple imports that bring in multiple symbols with the same name.  That should be an error case, but I don't think is.  At the very least, I'd like a test for that.

Reply:
Point 2 should really be in the ResolveScope file, but it made more sense as a full though with the other part to me.

Reply:
Can your 1st point be about a confusion with my function names? This function that is added here looks for public imports that enable unqualified access. But the cases that are captured by this addition are those that are qualified through a module that has an import in it.

Reply:
I'm referring to the placement of this if statement after the previous one (which as I understand it only returns modules that were directly imported).  Does that make sense?  We should do a quick call tomorrow if not :)

Reply:
Call sounds good

Reply:
For 1: A test that might help the discussion is [this one](https://github.com/chapel-lang/chapel/pull/15256/files#diff-d1c188cc6d547ef2f977c40ec7a8e6af). I think this is the right behavior?
For 2: I am wondering if [this test's](https://github.com/chapel-lang/chapel/pull/15256/files#diff-711ec1c89e298f42a920ea3dab8f0c90) behavior answers the concern.